filecoin/retriever.go

```diff
@@ -304,6 +309,7 @@ func (retriever *Retriever) retrieveFromBestCandidate(ctx context.Context, retri
 			stats.Record(ctx, metrics.RetrievalDealDuration.M(retrievalStats.Duration.Seconds()))
 			stats.Record(ctx, metrics.RetrievalDealSize.M(int64(retrievalStats.Size)))
 			stats.Record(ctx, metrics.RetrievalDealCost.M(retrievalStats.TotalPayment.Int64()))
+			stats.Record(ctx, metrics.FailedRetrievalsPerRequestCount.M(int64(i)))
```

Comment:
needs a duplicate of this down below in the `if retrievalStats == nil {` block - because if they all fail we need to see that too

Reply:
I think I talked through the dilemma on this one—on one hand, it's interesting to be able to ask "how many failures did we get before we got a success"—and in that respect the current implementation answers that directly because it doesn't count the all-failed case. But on the other hand, that's not really all that interesting in comparison to the "how many failures are we getting per request overall" whether it ends in a success or not.

I think we'll figure out what's really useful here when we deploy it but IIRC we came down on the side of "let's record this for all requests, regardless of wether we get a success or not".

Reply:
Ah, yeah, completely forgot this one.